### ISSUE TITLE
EgressQOSMaps, IngressQOSMaps map bug

### ISSUE THREAD

**Issue by glaive-tordu** [systemd/systemd#27460]
### systemd version the issue has been seen with

252

### Used distribution

Debian 11.7

### Linux kernel version used

5.10.0-22

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Unexpected behaviour you saw

Hello,

There is a bug with EgressQOSMaps and IngressQOSMaps, we cannot map SO_PRIORITY inferior to VLAN header PCP field.

From https://www.freedesktop.org/software/systemd/man/systemd.netdev.html,
"EgressQOSMaps=, IngressQOSMaps=

Defines a mapping of Linux internal packet priority (SO_PRIORITY) to VLAN header PCP field for outgoing and incoming frames, respectively. Takes a whitespace-separated list of integer pairs, where each integer must be in the range 1…4294967294, in the format "from"-"to", e.g., "21-7 45-5". Note that "from" must be greater than or equal to "to". When unset, the kernel's default will be used."

The command, ip link set "interface" type vlan ingress-qos-map|egress-qos-map, allow to map SO_PRIORITY inferior to PCP field.

Best regards.

**Comment by yuwata** [comment]
Sorry for late response. But I cannot follow the report. Could you elaborate more?
- Please provide .netdev file you use.
- Please show the result of `ip -d link show dev INTERFACE` when the vlan interface is created by networkd.
- Please also show the result of the expected result of `ip` command.
- Then, clarify which field is different.

**Comment by charles-martel55** [comment]
Hi,

```
[NetDev]
Name=test.101
Kind=vlan

[VLAN]
Id=101
EgressQOSMaps=0-1 1-3 10-3 6-6 7-7
```
Errors in unit journal :
```
systemd-networkd[1623]: /etc/systemd/network/test.101.netdev:7: Invalid EgressQOSMaps, ignoring: 0-1
systemd-networkd[1623]: /etc/systemd/network/test.101.netdev:7: Invalid EgressQOSMaps, ignoring: 1-3
```
Result :
```
ip -d link show dev test.101
11: test.101@enp6s19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether e6:28:59:68:da:ab brd ff:ff:ff:ff:ff:ff promiscuity 0  allmulti 0 minmtu 0 maxmtu 65535
    vlan protocol 802.1Q id 101 <REORDER_HDR>
      egress-qos-map { 6:6 7:7 10:3 } addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 tso_max_size 65536 tso_max_segs 65535 gro_max_size 65536 gso_ipv4_max_size 65536 gro_ipv4_max_size 65536

```

Missing mappings 0:1 and 1:3.


With ip command is it working:
```
ip link set test.101 type vlan egress-qos-map 0:1 1:3
ip -d link show dev test.101
11: test.101@enp6s19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether e6:28:59:68:da:ab brd ff:ff:ff:ff:ff:ff promiscuity 0  allmulti 0 minmtu 0 maxmtu 65535
    vlan protocol 802.1Q id 101 <REORDER_HDR>
      egress-qos-map { 0:1 1:3 6:6 7:7 10:3 } addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 tso_max_size 65536 tso_max_segs 65535 gro_max_size 65536 gso_ipv4_max_size 65536 gro_ipv4_max_size 65536
```

Best regards,

**Comment by yuwata** [comment]
Thank you!
Fix is waiting in #27761.
